benchmark/single-source/KeyPathPerformanceTests.swift

```diff
@@ -0,0 +1,1009 @@
+    case 55:
+      return \FixedSizeArray100.element55
+    case 56:
+      return \FixedSizeArray100.element56
+    case 57:
+      return \FixedSizeArray100.element57
+    case 58:
+      return \FixedSizeArray100.element58
+    case 59:
+      return \FixedSizeArray100.element59
+    case 60:
+      return \FixedSizeArray100.element60
+    case 61:
+      return \FixedSizeArray100.element61
+    case 62:
+      return \FixedSizeArray100.element62
+    case 63:
+      return \FixedSizeArray100.element63
+    case 64:
+      return \FixedSizeArray100.element64
+    case 65:
+      return \FixedSizeArray100.element65
+    case 66:
+      return \FixedSizeArray100.element66
+    case 67:
+      return \FixedSizeArray100.element67
+    case 68:
+      return \FixedSizeArray100.element68
+    case 69:
+      return \FixedSizeArray100.element69
+    case 70:
+      return \FixedSizeArray100.element70
+    case 71:
+      return \FixedSizeArray100.element71
+    case 72:
+      return \FixedSizeArray100.element72
+    case 73:
+      return \FixedSizeArray100.element73
+    case 74:
+      return \FixedSizeArray100.element74
+    case 75:
+      return \FixedSizeArray100.element75
+    case 76:
+      return \FixedSizeArray100.element76
+    case 77:
+      return \FixedSizeArray100.element77
+    case 78:
+      return \FixedSizeArray100.element78
+    case 79:
+      return \FixedSizeArray100.element79
+    case 80:
+      return \FixedSizeArray100.element80
+    case 81:
+      return \FixedSizeArray100.element81
+    case 82:
+      return \FixedSizeArray100.element82
+    case 83:
+      return \FixedSizeArray100.element83
+    case 84:
+      return \FixedSizeArray100.element84
+    case 85:
+      return \FixedSizeArray100.element85
+    case 86:
+      return \FixedSizeArray100.element86
+    case 87:
+      return \FixedSizeArray100.element87
+    case 88:
+      return \FixedSizeArray100.element88
+    case 89:
+      return \FixedSizeArray100.element89
+    case 90:
+      return \FixedSizeArray100.element90
+    case 91:
+      return \FixedSizeArray100.element91
+    case 92:
+      return \FixedSizeArray100.element92
+    case 93:
+      return \FixedSizeArray100.element93
+    case 94:
+      return \FixedSizeArray100.element94
+    case 95:
+      return \FixedSizeArray100.element95
+    case 96:
+      return \FixedSizeArray100.element96
+    case 97:
+      return \FixedSizeArray100.element97
+    case 98:
+      return \FixedSizeArray100.element98
+    case 99:
+      return \FixedSizeArray100.element99
+    default:
+      fatalError()
+    }
+  }
+
+  // Conformance to Iterator.
+  mutating func next() -> Element? {
+    var iter_count = 0
+    if iter_count == count {
+      return nil
+    } else {
+      defer { iter_count += 1 }
+      return self[keyPath: FixedSizeArray100.getKeypathToElement(index: iter_count)]
+    }
+  }
+}
+
+let kp0 = FixedSizeArray100<ElementType>.getKeypathToElement(index: 0)
```

Comment:
It's not good to keep the keypaths in global variables, because accessing (non-trivial) globals is slow.
Instead I recommend to create them in the benchmark functions (before the loop) and wrap them in `identity()` so that the optimizer does not constant propagate them.

Reply:
I've moved pertinent items into the benchmarks themselves. (Earlier on, we were getting warnings about setup times being too high, but I largely thing that was due to not using `n` correctly within the benchmark for-loops.)